### ISSUE TITLE
Table.reverse() fails if there are mixin columns that do not support assignment

### ISSUE THREAD

**Issue by ramonaaugustin** [astropy/astropy#9836]
### Description

Table cannot be reversed if it contains a SkyCoord.
### Expected behavior

I expected the table to reverse using `t.reverse()`
### Actual behavior

Table cannot be reversed if it contains a SkyCoord.
### Steps to Reproduce

I'm loading `my_photometry.txt` from the [photutils section of the astropy workshop at the AAS winter meeting 2020](https://github.com/astropy/astropy-workshop/blob/master/09-Photutils/photutils_overview.ipynb).

```python
In [1]: from astropy.table import QTable                                        

In [2]: tbl = QTable.read('my_photometry.txt', format='ascii.ecsv')             

In [3]: tbl                                                                     
Out[3]: 
<QTable length=3>
  id  xcenter ycenter ...      ra_icrs           dec_icrs     
        pix     pix   ...        deg               deg        
int64 float64 float64 ...      float64           float64      
----- ------- ------- ... ----------------- ------------------
    1   90.73   59.43 ... 53.15888794969764 -27.76499923317225
    2   73.63  139.41 ... 53.15920958492472 -27.76366614996854
    3   43.62   61.63 ... 53.15977526527222 -27.76496233290131

In [4]: tbl.reverse()                                                           
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-49611a028856> in <module>
----> 1 tbl.reverse()

~/opt/anaconda3/envs/astropy-workshop/lib/python3.7/site-packages/astropy/table/table.py in reverse(self)
   2972         '''
   2973         for col in self.columns.values():
-> 2974             col[:] = col[::-1]
   2975         for index in self.indices:
   2976             index.reverse()

TypeError: 'SkyCoord' object does not support item assignment
```
### System Details

Darwin-18.7.0-x86_64-i386-64bit
Python 3.7.5 (default, Oct 25 2019, 10:52:18) 
[Clang 4.0.1 (tags/RELEASE_401/final)]
Numpy 1.17.4
Scipy 1.3.1
astropy 4.0

EDIT: Formatting.

**Comment by pllim** [comment]
The error looks similar to #9536 that was fixed in #9549 , but maybe that patch did not address reverse sorting? cc @taldcroft and @mhvk

**Comment by pllim** [comment]
p.s. Isn't the `reverse` method redundant, as `sort` takes a `reverse` keyword anyway?

**Comment by taldcroft** [comment]
Indeed the fix for #9549 should have been applied to `reverse()` as well.

There is a simple workaround, namely:
```
tbl = tbl[::-1]
```

PR to fix this coming.

**Comment by pllim** [comment]
Thank you!
